cmd/sentry/download/sentry.go

```diff
@@ -904,156 +878,137 @@ func (ss *SentryServerImpl) GetStatus() *proto_sentry.StatusData {
 	return ss.statusData
 }
 
-func (ss *SentryServerImpl) restartReceive() {
+func (ss *SentryServerImpl) send(msgID proto_sentry.MessageId, peerID string, b []byte) {
+	ss.lock.RLock()
+	defer ss.lock.RUnlock()
+	errs := ss.streams[msgID].Broadcast(&proto_sentry.InboundMessage{
+		PeerId: gointerfaces.ConvertBytesToH512([]byte(peerID)),
+		Id:     msgID,
+		Data:   b,
+	})
+	for _, err := range errs {
+		log.Error("Sending msg to core P2P failed", "msg", proto_sentry.MessageId_name[int32(msgID)], "error", err)
+	}
+}
+
+func (ss *SentryServerImpl) hasSubscribers(msgID proto_sentry.MessageId) bool {
+	ss.lock.RLock()
+	defer ss.lock.RUnlock()
+	return ss.streams[msgID] != nil && ss.streams[msgID].Len() > 0
+	//	log.Error("Sending msg to core P2P failed", "msg", proto_sentry.MessageId_name[int32(streamMsg.msgId)], "error", err)
+}
+
+func (ss *SentryServerImpl) addStream(ids []proto_sentry.MessageId, server proto_sentry.Sentry_MessagesServer) func() {
 	ss.lock.Lock()
 	defer ss.lock.Unlock()
-	// Close previous channel and recreate
-	close(ss.stopCh)
-	ss.stopCh = make(chan struct{})
-}
+	if ss.streams == nil {
+		ss.streams = map[proto_sentry.MessageId]*StreamsList{}
+	}
 
-func (ss *SentryServerImpl) ReceiveMessages(_ *emptypb.Empty, server proto_sentry.Sentry_ReceiveMessagesServer) error {
-	ss.restartReceive()
 	checksumCache, err := lru.New(1024)
```

Comment:
Perhaps remove checkSumCache for now, because it is not compatible with some of the streams (like request streams)

Reply:
I need to find a different way of doing it without causing issues in other places anyway